plugins/pulp_docker/plugins/importers/importer.py

```diff
@@ -180,106 +184,148 @@ def import_units(self, source_repo, dest_repo, import_conduit, config, units=Non
             for u in units:
-                if u.type_id != constants.IMAGE_TYPE_ID:
-                    continue
-                conduit.associate_unit(u)
-                units_added.append(u)
+                repository.associate_single_unit(dest_repo, u)
+                units_added.add(u)
                 known_units.add(u.unit_key['image_id'])
-                parent_id = u.metadata.get('parent_id')
+                parent_id = u.parent_id
                 if parent_id:
                     units_to_add.add(parent_id)
             # Filter out units we have already added
             units_to_add.difference_update(known_units)
             # Find any new units to add to the repository
             if units_to_add:
-                unit_filter = {'image_id': {'$in': list(units_to_add)}}
-                criteria = UnitAssociationCriteria(type_ids=[constants.IMAGE_TYPE_ID],
-                                                   unit_filters=unit_filter)
-                units = conduit.get_source_units(criteria=criteria)
+                units = models.Image.objects.filter(image_id__in=list(units_to_add))
             else:
                 # Break out of the loop since there were no units to add to the list
                 break
 
-        return units_added
+        return list(units_added)
 
     @staticmethod
-    def _import_manifests(conduit, units):
+    def _import_tag(conduit, unit, dest_repo):
         """
-        Import manifests and referenced blobs.
+        Import a Tag, and the Manifests and Blobs it references.
 
-        :param conduit: provides access to relevant Pulp functionality
-        :type conduit: pulp.plugins.conduits.unit_import.ImportUnitConduit
-        :param units: optional list of pre-filtered units to import
-        :type  units: list of pulp.plugins.model.Unit
-        :return: list of units that were copied to the destination repository
-        :rtype:  list
+        :param conduit:   provides access to relevant Pulp functionality
+        :type  conduit:   pulp.plugins.conduits.unit_import.ImportUnitConduit
+        :param unit:      The Tag to be imported to the repository
+        :type  unit:      pulp_docker.plugins.models.Tag
+        :param dest_repo: The destination repository that the Tag is being imported to. This is
+                          needed because technically we are creating a copy of the Tag there rather
+                          than an association of the Tag, and the repo_id is a required field on the
+                          Tag object.
+        :type  dest_repo: pulp.plugins.model.Repository
+        :return:          list of Units that were copied to the destination repository
+        :rtype:           list
         """
-        units_added = []
+        units_added = set()
 
-        # All manifests if not specified
+        # We need to create a copy of the Tag with the destination repository's id, but other fields
+        # copied from the source Tag.
```

Comment:
This could be useful to have on the base ``ContentUnit`` for occasions like this: https://github.com/pulp/pulp_rpm/blob/b22a982e85a12edd42359d55e125ff6896989642/plugins/pulp_rpm/plugins/db/models.py#L49

It's not a big deal here though since the tag has so few attributes.